compiler-core/src/language_server/code_action.rs

```diff
@@ -7499,3 +7508,190 @@ impl<'ast> ast::visit::Visit<'ast> for RemoveBlock<'ast> {
+                variables_names,
+            });
+        }
+    }
+}
+
+pub struct UnwrapAnonymousFunction<'a> {
+    module: &'a Module,
+    line_numbers: &'a LineNumbers,
+    params: &'a CodeActionParams,
+    targets: Vec<FnToUnwrap>,
+}
+
+struct FnToUnwrap {
+    fn_location: SrcSpan,
+    inner_function_location: SrcSpan,
+}
+
+impl<'a> UnwrapAnonymousFunction<'a> {
+    pub fn new(
+        module: &'a Module,
+        line_numbers: &'a LineNumbers,
+        params: &'a CodeActionParams,
+    ) -> Self {
+        Self {
+            module,
+            line_numbers,
+            params,
+            targets: vec![],
+        }
+    }
+
+    pub fn code_actions(mut self) -> Vec<CodeAction> {
+        self.visit_typed_module(&self.module.ast);
+
+        let mut actions = Vec::with_capacity(self.targets.len());
+        for target in self.targets {
+            let mut edits = TextEdits::new(self.line_numbers);
+
+            edits.delete(SrcSpan {
+                start: target.fn_location.start,
+                end: target.inner_function_location.start,
+            });
+            edits.delete(SrcSpan {
+                start: target.inner_function_location.end,
+                end: target.fn_location.end,
+            });
+
+            CodeActionBuilder::new("Unwrap anonymous function")
+                .kind(CodeActionKind::REFACTOR_REWRITE)
+                .changes(self.params.text_document.uri.clone(), edits.edits)
+                .push_to(&mut actions);
+        }
+        actions
+    }
+}
+
+impl<'ast> ast::visit::Visit<'ast> for UnwrapAnonymousFunction<'ast> {
+    fn visit_typed_expr_fn(
+        &mut self,
+        location: &'ast SrcSpan,
+        _type_: &'ast Arc<Type>,
+        kind: &'ast FunctionLiteralKind,
+        arguments: &'ast [TypedArg],
+        body: &'ast Vec1<TypedStatement>,
+        _return_annotation: &'ast Option<ast::TypeAst>,
+    ) {
+        match kind {
+            FunctionLiteralKind::Anonymous { .. } => (),
+            _ => return,
+        }
+
+        // We need the existing argument list for the fn to be a 1:1 match for the args we pass
+        // to the called function. We figure out what the call-arg list needs to look like here,
+        // and bail out if any incoming args are discarded.
+        let mut expected_arguments = Vec::with_capacity(arguments.len());
+        for a in arguments {
+            match &a.names {
+                ArgNames::Named { name, .. } => expected_arguments.push(name),
+                ArgNames::NamedLabelled { name, .. } => expected_arguments.push(name),
+                ArgNames::Discard { .. } => return,
+                ArgNames::LabelledDiscard { .. } => return,
+            }
+        }
+
+        // match fn bodies with only a single function call
+        if let [stmt] = body.as_slice()
+            && let TypedStatement::Expression(expr) = stmt
+            && let TypedExpr::Call { fun, arguments, .. } = expr
```

Comment:
I think this is the first use in the repo of Rust 1.88's new chained lets. I saw in another ticket that you don't try to support old Rusts, so I hope this is OK. It looks a lot more readable to me than having the whole nested pattern on one line but I don't have a lot of Rust experience so it could just be unfamiliarity